account_wallet/models/account_move_line.py

```diff
@@ -0,0 +1,102 @@
+# © 2015  Laetitia Gangloff, Acsone SA/NV (http://www.acsone.eu)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools import float_compare
+from odoo.tools.translate import _
+from odoo.osv.expression import AND
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    account_wallet_id = fields.Many2one(
+        comodel_name='account.wallet',
+        string='Wallet',
+        ondelete='restrict',
+        index=True,
+    )
+
+    def _get_computed_account(self):
+        if self.account_wallet_id:
+            return self.account_wallet_id.wallet_type_id.account_id
+        return super()._get_computed_account()
+
+    @api.onchange("account_wallet_id")
+    def _onchange_account_wallet_id(self):
+        for line in self.filtered("account_wallet_id"):
+            line.account_id = line._get_computed_account()
+            partner = line.account_wallet_id.partner_id
+            if partner:
+                line.partner_id = partner
```

Comment:
can we have different partner in move line than the one in move level?

Reply:
Mmmmh, I think this has been done to cover the case of accounting people writing OD move with several partners at one time in case of correction.